iOS8-HW11-Yuliya-Khrebtova/Sources/ViewController.swift

```diff
@@ -0,0 +1,299 @@
+        twitterButton.layer.shadowOffset = .zero
+        twitterButton.layer.shadowRadius = 5
+        twitterButton.layer.shouldRasterize = true
+        twitterButton.layer.rasterizationScale = UIScreen.main.scale
+        twitterButton.translatesAutoresizingMaskIntoConstraints = false
+        return twitterButton
+    }()
+
+    private lazy var socialMediaButtons: UIStackView = {
+        let socialMediaButtons = UIStackView()
+        socialMediaButtons.axis = .horizontal
+        socialMediaButtons.alignment = .center
+        socialMediaButtons.spacing = 20
+        socialMediaButtons.addArrangedSubview(facebookButton)
+        socialMediaButtons.addArrangedSubview(twitterButton)
+        socialMediaButtons.translatesAutoresizingMaskIntoConstraints = false
+        return socialMediaButtons
+    }()
+
+    private lazy var dontHaveAccountLabel: UILabel = {
+        let dontHaveAccountLabel = UILabel()
+        dontHaveAccountLabel.text = "Don't have account?"
+        dontHaveAccountLabel.textColor = .gray
+        dontHaveAccountLabel.textAlignment = .center
+        dontHaveAccountLabel.font = UIFont.boldSystemFont(ofSize: 16)
+        dontHaveAccountLabel.translatesAutoresizingMaskIntoConstraints = false
+        return dontHaveAccountLabel
+    }()
+
+    private lazy var signUpButton: UIButton = {
+        let signUpButton = UIButton()
+        signUpButton.setTitle("Sign up", for: .normal)
+        signUpButton.titleLabel?.font = UIFont.boldSystemFont(ofSize: 16)
+        signUpButton.setTitleColor(UIColor.systemIndigo, for: .normal)
+        signUpButton.translatesAutoresizingMaskIntoConstraints = false
+        return signUpButton
+    }()
+
+    private lazy var signUpStack: UIStackView = {
+        let signUpStack = UIStackView()
+        signUpStack.axis = .horizontal
+        signUpStack.alignment = .center
+        signUpStack.spacing = 20
+        signUpStack.addArrangedSubview(dontHaveAccountLabel)
+        signUpStack.addArrangedSubview(signUpButton)
+        signUpStack.translatesAutoresizingMaskIntoConstraints = false
+        return signUpStack
+    }()
+
+    // MARK: - Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+        setupHierarchy()
+        setupLayout()
+    }
+
+    // MARK: - Setup
+
+    private func setupHierarchy() {
+        view.addSubview(background)
+        view.addSubview(label)
+        view.addSubview(loginInput)
+        view.addSubview(passwordInput)
+        view.addSubview(loginButton)
+        view.addSubview(forgotPasswordButton)
+        view.addSubview(connectLabel)
+        view.addSubview(leftLine)
+        view.addSubview(rightLine)
+        view.addSubview(socialMediaButtons)
+        view.addSubview(signUpStack)
+    }
+
+    private func setupLayout() {
+        label.snp.makeConstraints { make in
+            make.centerX.equalTo(view)
+            make.top.equalTo(view).offset(70)
+        }
```

Comment:
Тут можно упростить вот таким образом. Но в целом я бы рекомендовал учиться на констрейнтах, которые дает Apple так будет больше понимания, а Snapkit потом освоить не составит труда. Остальное по аналогии
```suggestion
        label.snp.makeConstraints {
            $0.centerX.equalToSuperview()
            $0.top.equalToSuperview().offset(70)
        }
```